pkg/asset/installconfig/gcp/dns.go

```diff
@@ -34,41 +40,93 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
+	client, err := NewClient(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	dnsZone, err := client.GetPublicDNSZoneByID(ctx, project, id)
+	if err != nil {
+		return nil, err
+	}
+	return dnsZone, nil
+}
+
 // GetBaseDomain returns a base domain chosen from among the project's public DNS zones.
-func GetBaseDomain(project string) (string, error) {
+func GetBaseDomain(project string) (*Zone, error) {
 	client, err := NewClient(context.TODO())
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
 	defer cancel()
 
-	publicZones, err := client.GetPublicDomains(ctx, project)
+	publicZonesMap, err := client.GetPublicDomains(ctx, project)
 	if err != nil {
-		return "", errors.Wrap(err, "could not retrieve base domains")
+		return nil, errors.Wrap(err, "could not retrieve base domains")
+	}
+
+	publicZones := make([]string, 0, len(publicZonesMap))
+	for name := range publicZonesMap {
+		publicZones = append(publicZones, name)
 	}
+
 	if len(publicZones) == 0 {
-		return "", errors.New("no domain names found in project")
+		return nil, errors.New("no domain names found in project")
 	}
 	sort.Strings(publicZones)
 
-	var domain string
+	var publicZoneName string
+	var publicZoneID string
+
 	if err := survey.AskOne(&survey.Select{
 		Message: "Base Domain",
 		Help:    "The base domain of the cluster. All DNS records will be sub-domains of this base and will also include the cluster name.\n\nIf you don't see you intended base-domain listed, create a new public hosted zone and rerun the installer.",
 		Options: publicZones,
-	}, &domain, func(ans interface{}) error {
+	}, &publicZoneName, func(ans interface{}) error {
 		choice := ans.(string)
 		i := sort.SearchStrings(publicZones, choice)
 		if i == len(publicZones) || publicZones[i] != choice {
 			return errors.Errorf("invalid base domain %q", choice)
 		}
 		return nil
 	}); err != nil {
-		return "", errors.Wrap(err, "failed UserInput for base domain")
+		return nil, errors.Wrap(err, "failed UserInput for base domain")
+	}
+
+	zoneList, ok := publicZonesMap[publicZoneName]
+	if !ok {
+		return nil, errors.Errorf("invalid base domain %q", publicZoneName)
+	}
+
+	zoneListLen := len(zoneList)
+	if zoneListLen == 0 {
+		return nil, errors.Wrap(err, "there must be at least one public zones")
+	}
+	if zoneListLen == 1 {
+		return &Zone{
+			ID:   zoneList[0],
+			Name: publicZoneName,
+		}, nil
+	}
+
+	err = survey.AskOne(&survey.Select{
+		Message: "Zone ID",
+		Help:    "The zone ID for the base domain of the cluster",
+		Options: zoneList,
+	}, &publicZoneID, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed UserInput for zone ID")
 	}
 
-	return domain, nil
+	return &Zone{
+		ID:   publicZoneID,
+		Name: publicZoneName,
+	}, nil
```

Comment:
nit: maybe just make it a single prompt?

```
test.cluster.com (zone-1)
test.cluster.com (zone-2)
prd.cluster.com (zone-3)
```

Reply:
Updated!